### PR TITLE
Add ffi for reorder_visual

### PR DESCRIPTION
**Pull request by Manishearth**

### CODE REVIEW

File: ffi/diplomat/js/include/ICU4XBidi.d.ts [hunk 1]
```diff
@@ -31,6 +32,16 @@ export class ICU4XBidi {
    */
   for_text(text: string, default_level: u8): ICU4XBidiInfo;
 
+  /**
+
+   * Utility function for producing reorderings given a list of levels
+
+   * Produces a map saying which visual index maps to which source index.
+
+   * See the {@link https://unicode-org.github.io/icu4x-docs/doc/unicode_bidi/struct.Level.html#method.is_rtl Rust documentation for `is_rtl`} for more information.
```
Comment (sffc) [inline ffi/diplomat/js/include/ICU4XBidi.d.ts:41]: Observation: This link is broken

File: ffi/diplomat/src/bidi.rs [hunk 2]
```diff
@@ -60,6 +61,14 @@ pub mod ffi {
                 Level::new(default_level).ok(),
             )))
         }
+        /// Utility function for producing reorderings given a list of levels
+        ///
+        /// Produces a map saying which visual index maps to which source index.
+        #[diplomat::rust_link(unicode_bidi::Level::is_rtl, FnInStruct)]
+        pub fn reorder_visual(&self, levels: &[u8]) -> Box<ICU4XReorderedIndexMap> {
+            let levels = Level::from_slice_unchecked(levels);
```
Comment (sffc) [inline ffi/diplomat/src/bidi.rs:69]: Question: is it safe to use a method named "unchecked" here in this context where you definitely haven't performed any checking on the &[u8] ?

File: ffi/diplomat/src/bidi.rs [hunk 3]
```diff
@@ -60,6 +61,14 @@ pub mod ffi {
                 Level::new(default_level).ok(),
             )))
         }
+        /// Utility function for producing reorderings given a list of levels
+        ///
+        /// Produces a map saying which visual index maps to which source index.
+        #[diplomat::rust_link(unicode_bidi::Level::is_rtl, FnInStruct)]
```
Comment (sffc) [inline ffi/diplomat/src/bidi.rs:67]: Nit: This should link to the `reorder_visual` function, not `is_rtl`

**Review by sffc — COMMENTED** [review]
I assume this is needed by CanvasKit/Skia? I had been getting mixed messages for whether they needed this function. But it seems harmless to export it anyway.

**Comment by Manishearth** [comment]
Yes. I don't like this function because it needs the caller to manage levels.

But skia wants to do that and we shouldn't stop them, it's not bad it's just clunky (and it is more powerful).

**Review by sffc — APPROVED** [review]

**Pull request merged**